Improve pagination documentation (#9488)
withastro/docs: src/content/docs/en/reference/api-reference.mdx

It can also be used in static file endpoints for [dynamic routing](/en/guides/endpoints/#params-and-dynamic-routing).

:::tip

> When using TypeScript, use the [`InferGetStatic*`](/en/guides/typescript/#infer-getstaticpaths-types) type utilities to ensure type-safe access of your `params` and `props`.

Reviewer: I'm a little late, I was waiting until the PR was no longer a draft to comment.

`InferGetStatic` does not exist and I don't think the asterisk is understandable to everyone. It seems to me that `GetStaticPaths` is enough to have both `Astro.params` and `Astrop.props` inferred properly. The two other helpers (`InferGetStaticParamsType` and `InferGetStaticPropsType`) can be useful in some cases but I don't think they are required here.
So I would suggest:

```suggestion
When using TypeScript, use the [`GetStaticPaths`](/en/guides/typescript/#infer-getstaticpaths-types) type utility to ensure type-safe access of your `params` and `props`.
```

Author: Thanks for the suggestion! How about the following phrasing:
```md
When using TypeScript, follow the [Infer `getStaticPaths` types](/en/guides/typescript/#infer-getstaticpaths-types) guide to ensure type-safe access of your `params` and `props`.
```

Reviewer: Why not, all types are useful. If I insisted on `GetStaticPaths` it is perhaps because I find that this section of the Typescript guide seems to insist on the other two types (`InferGetStaticParamsType` and `InferGetStaticPropsType`) when only one (`GetStaticPaths`) is enough for most cases. But that might just be my perception...
So I'm fine with your suggestion, if you think it's more helpful!

Author: After looking into it, it appears that GetStaticPaths is sufficient for type safety. However, this raises the question of why there are additional type utilities included. Additionally, that section does not highlight the line where we add satisfies GetStaticPaths; it only mentions the InferGetStatic* types. It seems like there’s a new area for improvement here?Tthough I believe it falls outside the scope of this PR.

Considering this, I will go with your suggestion. It seems satisfactory.

Reviewer: Um yes I think there is a missing line among those highlighted. Maybe it was forgotten because the opening parenthesis is also important in this case. As for the usefulness of the other types... I don't know. I've never used them, but no doubt some people need them. Better excess types than missing types! 😄 
But you're right, it doesn't fit into this PR.

> :::tip
> When using TypeScript, use the [`GetStaticPaths`](/en/guides/typescript/#infer-getstaticpaths-types) type utility to ensure type-safe access of your `params` and `props`.
> :::

Reviewer 2: ```suggestion
```
I agree with the comment from the issue about NOT putting this here. This is about `getStaticPaths()` and its type should be documented there.

Also, it's not a "tip" if it's "how things work."  😄

const { page } = Astro.props;
```

`paginate()` has the following arguments:

> - `data` - Array of data to paginate

Reviewer 2: ```suggestion
- `data` - array containing the page’s data passed to the `paginate()` function
```

> - `options` - Optional object with following properties:

Reviewer: ```suggestion
- `options` - Optional object with following the properties:
```
